### PR TITLE
Added Collection Parameters support for the teamcity generation

### PR DESCRIPTION
**Pull request by chrsin**
I confirm that the pull-request:

- [X ] Follows the contribution guidelines
- [X ] Is based on my own work
- [X ] Is in compliance with my employer

### CODE REVIEW

File: source/Nuke.Common.Tests/CI/ConfigurationGenerationTest.cs [hunk 1]
```diff
@@ -216,6 +216,24 @@ yield return
             [Parameter("Azure Pipelines System Access Token")]
             public readonly string AzurePipelinesSystemAccessToken;
 
+            [Parameter("Test collections work as input")]
```
Comment (matkoch) [inline source/Nuke.Common.Tests/CI/ConfigurationGenerationTest.cs:219]: Actually, `Parameter` only supports arrays.

File: source/Nuke.Common/CI/TeamCity/TeamCityAttribute.cs [hunk 2]
```diff
@@ -285,14 +286,20 @@ TeamCityParameterType GetParameterType()
                     return TeamCityParameterType.Select;
                 return TeamCityParameterType.Text;
             }
-
+            
+            var defValue = defaultValue?.ToString();
```
Comment (matkoch) [inline source/Nuke.Common/CI/TeamCity/TeamCityAttribute.cs]: Ternary conditional operator with reassignment.
`coll` => `defaultValues`

File: source/Nuke.Common.Tests/CI/ConfigurationGenerationTest.cs (hunk 1 above)
Comment (chrsin) [reply source/Nuke.Common.Tests/CI/ConfigurationGenerationTest.cs:219]: Removed values that aren't Array

File: source/Nuke.Common/CI/TeamCity/TeamCityAttribute.cs (hunk 2 above)
Comment (chrsin) [reply source/Nuke.Common/CI/TeamCity/TeamCityAttribute.cs]: Done

**Comment by matkoch** [comment]
`allowMultiple` seems only to be valid with a list of options

**Pull request closed without merging**